interp/constrextern.ml

```diff
@@ -307,8 +307,20 @@ let extern_evar n l = CEvar (n,l)
     For instance, in the debugger the tables of global references
     may be inaccurate *)
 
+let rec dirpath_of_modpath = function
+  | MPfile dp -> dp
+  | MPbound mbid -> let (_,id,_) = MBId.repr mbid in DirPath.make [id]
+  | MPdot (t, l) -> Libnames.add_dirpath_suffix (dirpath_of_modpath t) (Label.to_id l)
+
+let path_of_global = function
+  | GlobRef.VarRef id -> Libnames.make_path DirPath.empty id
+  | GlobRef.ConstRef cst -> Libnames.make_path (dirpath_of_modpath (Constant.modpath cst)) (Label.to_id (Constant.label cst))
+  | GlobRef.IndRef (ind,_) -> Libnames.make_path (dirpath_of_modpath (MutInd.modpath ind)) (Id.of_string ("<unknown-inductive>" ^ Label.to_string (MutInd.label ind) ^ ">"))
```

Comment:
There is a typo here in the string, this should read `"<unknown-inductive-"`.

Reply:
Right. I also forgot that `<` will be rejected by `Id.of_string`. So, I created a valid ident instead.